- Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [1]:
sigma = 16
M = 80
n = 256
alpha = 1 - 0.95
z = 1.96

Рассчитаем предельное отклонение от средней арифметической по следующей формуле:

$$T_{1,2} = \overline{X} \pm \frac{s_0}{\sqrt{n}} \cdot c_{\gamma}$$

где $T_1$, $T_2$ — нижняя и верхняя границы доверительного интервала,

$\overline{X}$  — выборочное среднее арифметическое, $s_0$ — среднее квадратичное отклонение по выборке (несмещенное),

$n$ — размер выборки,

$\gamma$ — доверительная вероятность,

$c_{\gamma} = Ф^{-1} \frac{(1 + \gamma)}{2}$ — обратное значение функции стандартного нормального распределения.

Итоговая формула: $$\left(\overline{x} - Z_{\frac{a}{2}} \cdot \frac{\sigma}{\sqrt{n}} , \overline{x} + Z_{\frac{a}{2}} \cdot \frac{\sigma}{\sqrt{n}}\right)$$

In [2]:
T1 = M - z * sigma/n ** 0.5
T2 = M + z * sigma/n ** 0.5
print(f'Доверительный интервал имеет границы ({T1}; {T2})')

Доверительный интервал имеет границы (78.04; 81.96)


2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [3]:
x = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
alpha = 1 - 0.95
t = 2.262
# mu = x mean
mu = sum(x) / len(x)
std = np.std(x, ddof=1)
T1 = mu - t * std / (len(x)) ** 0.5
T2 = mu + t * std / (len(x)) ** 0.5
print(f'Доверительный интервал имеет границы ({T1}; {T2})')

Доверительный интервал имеет границы (6.267538255912425; 6.912461744087573)


3, 4 задачи решать через тестирование гипотезы
3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

In [4]:
mu0 = 17
mu = 17.5
a = 0.05
n = 100
d = 4
zt = 1.645
zr = (mu-mu0)/(d ** 0.5)/(n ** 0.5)
print(f'Zр: {zr}')

Zр: 0.025


Поскольку Zрасчетное > Zтабличное, нулевая гипотеза отвергается. Верна гипотиза H1 при уровне значимости 5%

4. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [5]:
# H0: вес - 200 г, mu = mu0
# H1: вес - отличается от 200 г, mu не равно mu0
x = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
alpha = 1 - 0.99
n = len(x)
t_table = 3.250  # для alpha/2
mean_x = sum(x)/len(x)
mean_x
std_x = np.std(x,ddof=1)
t = (mean_x - 200) / (std_x / np.sqrt(n))
print(f't-критерий табличный = {t_table}\nt-критерий расчётный = {t}')

t-критерий табличный = 3.25
t-критерий расчётный = -1.0651074037450896


Поскольку t расчетное (-1.065) попадает в диаппазон tтабличное [-3.250;3.250], нулевая гипотеза подтверждается. Утверждение продавца верно.